src/datasources/email-api/pushwoosh-api.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { v4 as uuidv4 } from 'uuid';
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { HttpErrorFactory } from '@/datasources/errors/http-error-factory';
+import {
+  INetworkService,
+  NetworkService,
+} from '@/datasources/network/network.service.interface';
+import { CreateEmailMessageDto } from '@/domain/email/entities/create-email-message.dto.entity';
+import { IEmailApi } from '@/domain/interfaces/email-api.interface';
+
+@Injectable()
+export class PushwooshApi implements IEmailApi {
+  private readonly apiKey: string;
+  private readonly applicationCode: string;
+  private readonly baseUri: string;
+  private readonly fromEmail: string;
+  private readonly fromName: string;
+
+  constructor(
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+    @Inject(NetworkService)
+    private readonly networkService: INetworkService,
+    private readonly httpErrorFactory: HttpErrorFactory,
+  ) {
+    this.apiKey = this.configurationService.getOrThrow<string>('email.apiKey');
+    this.applicationCode = this.configurationService.getOrThrow<string>(
+      'email.applicationCode',
+    );
+    this.baseUri =
+      this.configurationService.getOrThrow<string>('email.baseUri');
+    this.fromEmail =
+      this.configurationService.getOrThrow<string>('email.fromEmail');
+    this.fromName =
+      this.configurationService.getOrThrow<string>('email.fromName');
+  }
+
+  /**
+   * Creates an email message through the Pushwoosh API.
+   * For details on the specification see:
+   * https://docs.pushwoosh.com/platform-docs/api-reference/email-api#createemailmessage
+   */
+  async createMessage(
+    createEmailMessageDto: CreateEmailMessageDto,
+  ): Promise<void> {
+    try {
+      const url = `${this.baseUri}/json/1.3/createEmailMessage`;
+      await this.networkService.post(url, {
+        request: {
+          application: this.applicationCode, // application code, should exist on Pushwoosh
+          auth: this.apiKey,
+          notifications: [
+            {
+              transactionId: uuidv4(), // unique identifier to avoid messages duplication
```

Comment:
I think this should be provided as an argument of `createMessage` or be part of `CreateEmailMessageDto`.

It would be up to the consumers of this API to know what is considered a unique event or not. By generating an id here, every call to `createMessage` will be considered unique.

Reply:
I was considering this internal to the provider, as an implementation detail. From an API consumer point of view, each email message should be meant to be sent once. Do you see a scenario where the consumer wants to allow the provider to send the same email message potentially N times?

Reply:
If I understand the provider documentation correctly, this is more to prevent the provider from sending the same message several times for no reason. I'm not even sure why this is a parameter in their API, seems like a mechanism to circumvent a malfunction on their side, not sure why this is not Pushwoosh internal 🤔

Reply:
I don't see it as internal because it's part of the public API. Else this would be generated on their side or?

Reply:
I meant internal to our `PushwooshApi` class, not the provider REST API.

But giving this a second thought, adding this unique identifier as a parameter as you suggested won't hurt, and it adds some degree of flexibility. I'm going to change this 👍

Reply:
I've added a field to give the clients the opportunity to send an unique identifier into `CreateEmailMessageDto` in [bd7cf93](https://github.com/safe-global/safe-client-gateway/pull/827/commits/bd7cf93077705307fe3e01efc758637736e90b31)